gamla/functional_generic.py

```diff
@@ -208,3 +205,45 @@ def composition_over_binary_curried(*args):
         pair_with,
     )
 )
+
+
+_first_truthy_index = compose_left(
+    enumerate,
+    curried.filter(toolz.second),
+    curried.map(toolz.first),
+    toolz.excepts(StopIteration, toolz.first),
+)
+
+
+class NoConditionMatched(Exception):
+    pass
+
+
+def _case(predicates: Tuple[Callable, ...], mappers: Tuple[Callable, ...]):
+    """Case with functions.
+
+    Handles async iff one of the predicates is async.
+    Raises `KeyError` if no condition matched.
+    """
+    return compose_left(
+        pair_right(
+            compose_left(
+                lazyjuxt(*predicates),
+                _first_truthy_index,
+                functional.check(
+                    toolz.complement(operator.eq(None)), NoConditionMatched
+                ),
+                mappers.__getitem__,
+            )
+        ),
+        functional.star(functional.apply),
+    )
+
+
+def case(predicates_and_mappers: Tuple[Tuple[Callable, ...], Tuple[Callable, ...]]):
```

Comment:
shouldn't the typing be Tuple[Tuple[Callable, Callable], ...]?

Reply:
No because it is exactly two elements.